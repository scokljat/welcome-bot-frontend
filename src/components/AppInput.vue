<template>
  <div class="input-box">
    <label
      data-testid="label"
      class="input-label"
      :class="{ 'error-label': error }"
      >{{ title }}</label
    >
    <input
      class="input-text"
      :class="{ 'error-input': error }"
      type="text"
      name="title"
      :placeholder="placeholder"
      :value="modelValue"
      @input="$emit('update:modelValue', $event.target.value)"
    />
    <span v-if="error" class="error-message">{{ error }}</span>
  </div>
</template>

<script>
export default {
  name: 'AppInput',
  props: {
    title: {
      type: String,
      required: true,
    },
    placeholder: {
      type: String,
      required: true,
    },
    modelValue: {
      type: String,
      required: true,
    },
    error: {
      type: String,
      default: null,
    },
  },
  emits: ['update:modelValue'],
};
</script>

<style lang="scss" scoped>
.input-box {
  @include input-box;
}

.input-label {
  @include input-label;
}

.input-text {
  @include input-text;
}
</style>
