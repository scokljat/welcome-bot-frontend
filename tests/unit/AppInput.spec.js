import { render, fireEvent } from '@testing-library/vue';
import '@testing-library/jest-dom';
import AppInput from '@/components/AppInput';

const props = {
  title: 'Message',
  placeholder: 'Enter the message...',
  modelValue: '',
  error: '',
};

const factory = (props) =>
  render(AppInput, {
    props,
  });

describe('AppInput.vue', () => {
  test('renders properly', () => {
    const wrapper = factory(props);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('checks input event', () => {
    const wrapper = factory(props);
    const input = wrapper.getByPlaceholderText('Enter the message...');
    fireEvent.change(input, { target: { value: 'New message' } });

    expect(input.value).toBe('New message');
  });

  test('renders error element', () => {
    props.error = 'Error occurred';
    const wrapper = factory(props);

    expect(wrapper.getByText('Error occurred'));
  });

  test('renders red label text', () => {
    props.error = 'Error occurred';
    const wrapper = factory(props);

    expect(wrapper.getByTestId('label')).toHaveClass('error-label');
  });

  test('renders red border around input', () => {
    props.error = 'Error occurred';
    const wrapper = factory(props);
    const input = wrapper.getByPlaceholderText('Enter the message...');

    expect(input).toHaveClass('error-input');
  });
});
