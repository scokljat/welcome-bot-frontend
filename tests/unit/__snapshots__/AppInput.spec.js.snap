// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput.vue renders properly 1`] = `
<div class="input-box"><label data-testid="label" class="input-label">Message</label><input class="input-text" type="text" name="title" placeholder="Enter the message...">
  <!--v-if-->
</div>
`;
